bin/desi_qso_mgii_afterburner

```diff
@@ -0,0 +1,250 @@
+                                  min_sigma, min_deltachi2, min_signifiance_A, min_A
+        DESI_TARGET (str): name of DESI_TARGET for the wanted version of the target selection
+        save_target (str) : restricted (save only IS_QSO_MGII==true targets) / all (save all the sample)
+    Returns:
+        QSO_sel (pandas dataframe): contains all the information useful to build the QSO cat
+    """
+
+    # just to speed up the process and don't go in mgii_fitter function for nothing..
+    if sel_to_mgii.sum() == 0:
+        QSO_sel = pd.DataFrame()
+    else:
+        (index_selected_with_mgii_fit,
+        fits_result,
+        index_with_mgii_fit) = mgii_fitter(spectra_name, zbest_name, sel_to_mgii,
+                                           param_mgii_fitter['lambda_width'],
+                                           max_sigma=param_mgii_fitter['max_sigma'],
+                                           min_sigma=param_mgii_fitter['min_sigma'],
+                                           min_deltachi2=param_mgii_fitter['min_deltachi2'],
+                                           min_A=param_mgii_fitter['min_A'],
+                                           min_signifiance_A=param_mgii_fitter['min_signifiance_A'])
+
+
+        sel_MGII = sel_to_mgii.copy()
+        # we only consider index where the mgii fit was done
+        sel_MGII[sel_to_mgii] = index_with_mgii_fit
+        # we then conserve only index where the mgii fit gives a good result !
+        sel_MGII[sel_MGII] = index_selected_with_mgii_fit
+
+        # Build dataframe to store the result
+        QSO_sel = pd.DataFrame()
+
+        if save_target == 'restricted':
+            index_to_save = sel_MGII.copy()
+            index_to_save_fit_result = index_with_mgii_fit.copy()
+            # keep only object selected by MGII
+            index_to_save_fit_result[index_with_mgii_fit] = index_selected_with_mgii_fit
+        elif save_target == 'all':
+            index_to_save = sel_to_mgii.copy()
+            # save every object with nan value if it is necessary --> there are sel_to_mgii.sum() objects to save
+            # index_with_mgii_fit is size of sel_to_mgii.sum()
+            index_to_save_fit_result = np.ones(sel_to_mgii.sum(), dtype=bool)
+        else:
+            log.error('**** CHOOSE CORRECT SAVE_TARGET FLAG (restricted / all) ****')
+
+        QSO_sel['TARGETID'] = zbest['TARGETID'][:][index_to_save]
+        QSO_sel['RA'] = fibermap['TARGET_RA'][:500][index_sorted][index_to_save]
+        QSO_sel['DEC'] = fibermap['TARGET_DEC'][:500][index_sorted][index_to_save]
+        QSO_sel['Z_RR'] = zbest['Z'][:][index_to_save]
+        QSO_sel['ZERR'] = zbest['ZERR'][:][index_to_save]
+        QSO_sel['COEFFS'] = zbest['COEFF'][:][index_to_save].tolist()
+        QSO_sel['Z_RR'] = zbest['Z'][:][index_to_save]
+        QSO_sel['SPECTYPE'] = zbest['SPECTYPE'][:][index_to_save]
+        QSO_sel[DESI_TARGET] = fibermap[DESI_TARGET][:500][index_sorted][index_to_save]
```

Comment:
Another case of a fragile `[:500]` that won't work in general in the future (at least for for healpix grouping).

Reply:
solved with new index_sorted